File: InterviewBit/pallindrome.java
public class pallindrome {
	public static boolean isPalindrome(int a) {
		
	    int i,j,len,fir,las,len2;
	    j=a;
	    i=0;
	    while(j>0)
	    {    j=j/10;    i++;    }
	    len2 = len = i;
	    System.out.println(len);
	    for(i=1;i<=len2/2; i++)
	    {
	        fir = a/(int)(Math.pow(10.0,(len-1)));
	        las = a%10;
	        System.out.println("a "+i+" :"+a);
	        System.out.println("len "+i+" :"+len);
	        System.out.println("first "+i+" :"+fir);
	        System.out.println("last "+i+" :"+las);
	        if(fir!= las)  return false;
	        a = a - fir*(int)(Math.pow(10.0,(len-1)));//minus
	        a = a/10;
	        len-=2;
	    }
	    return true;
	}
	public static void main(String[] args)
	{
		System.out.println(isPalindrome(123454321));
	}
}

File: InterviewBit/pallindrome.cpp
#include <iostream>
#include <cmath>

using namespace std;
int main()
	{
		int a,i;
		cin>>a;
	    int j,len,fir,las;
	    j=a;
	    i=0;
	    while(j>0)
	    {
	        j=j/10;		i++;
	    }
	    len = i;
	    cout<<"len ="<<len<<"   ";
	    for(i=1;i<=a/2;i++)
	    {
	        fir = (double)a/(pow(10.0,(--len)));
	        las = a%10;
	        if(fir!= las)  cout<<endl<<fir<<"  "<<las<<"   Not a Pallindrome";
	        a = a - (int)(fir*(pow(10.0,len+1)));//minus
	        a = a/10;
	        //len--;
	    }
	    cout<<"Pallindrome";
	    return 0;
	}

